protocol/git.md

```diff
@@ -0,0 +1,85 @@
+Git Protocol
+============
+
+A guide for programming within version control.
+
+Maintain a Repo
+---------------
+
+* Avoid including files in source control that are specific to your
+  development machine or process by adding their formats to the .gitignore file.
+* Delete local and remote feature branches after merging.
+* Perform all new work in a feature branch.
+* Rebase frequently to incorporate upstream changes.
+* Use a [pull request] for code reviews.
+
+[pull request]: https://help.github.com/articles/using-pull-requests/
+
+Write a Feature
+---------------
+
+For all new development work, create a local feature branch based off develop.
+Be sure to include the JIRA ticket in your feature name (example: DHH-123).
```

Comment:
This is a pattern that I've seen a few times recently.  On one hand, it clearly links it to the issue being addressed.  On the other hand, it's opaque and doesn't include an idea of what the branch is about.  Would it be too cumbersome to name branches feature/short-name-DHH-123?